Комментарий к 4му ДЗ:  
В третьей задаче я не использовал np.pi и np.exp, т.к. в условии сказано, что обойтись без сторонних библиотек np и пр.

In [8]:
import numpy as np
from scipy import stats

Задача 1

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.



In [3]:


h0 = 17
alpha = 0.025
n = 100
M = 17.5
sigma = 2
# Такой уровень значимости соответствует квантилю 2
# Ожидается, что это выборка с нормальным распределением

Tx = (M - h0)/(sigma/np.sqrt(n))
Tx

2.5

In [ ]:
Tx больше двух, поэтому данные противоречат гипотезе h0.

Задача 2

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания a с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.




In [5]:

alpha = 0.05
n = 256
M = 80
sigma = 16
# Снова у нас доверительный интервал с квантилем 2

#| Tx = (M - h0)/(sigma/np.sqrt(n))| < 2


#Тогда математическое ожидание должно быть в интервале
(M - 2 * sigma/np.sqrt(n), M + 2 * sigma/np.sqrt(n))

(78.0, 82.0)

Задача 3

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190

Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

In [27]:

weight = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
n = 10
M = weight.mean()
alpha = 0.01
std = weight.std(ddof=1)
t = stats.t.ppf(1 - alpha / 2, df=n - 1)
print(t)
print((M - h0)/(std/np.sqrt(n)))
(M - t * std / np.sqrt(n), M + t * std / np.sqrt(n))

3.2498355440153697
128.87799585315585


(193.92322840036354, 203.07677159963646)

200 попало в доверительный интервал (193.92322840036354, 203.07677159963646), считаем что продавец сказал правду.